hat/app/org/hatdex/hat/utils/AdjudicatorRequest.scala

```diff
@@ -50,49 +56,73 @@ object AdjudicatorRequestTypes {
   final case class ContractLeft(contractId: ContractId)
 }
 
-class AdjudicatorRequest(adjudicatorEndpoint: String, ws: WSClient) {
+class AdjudicatorRequest(adjudicatorEndpoint: String, sharedSecret: String, ws: WSClient) {
   import AdjudicatorRequestTypes._
 
-  def getPublicKey(hatName: String, contractId: ContractId, keyId: String)(implicit ec: ExecutionContext): Future[Either[PublicKeyRequestFailure, PublicKeyReceived]] = {
+  //
+  private val secret = JwtSecretKey(sharedSecret)
+  val hatClaimBuiler: JwtClaimBuilder[HatClaim] = HatClaim.builder
 
-    val url =
-      s"${adjudicatorEndpoint}/v1/contracts/${contractId}/hat/${hatName}/${keyId}"
-    val eventuallyResponse = makeRequest(url, ws).get()
+  // Create HatClaim
+  private def createHatClaim(hatNameAsStr: String, contractId: ContractId): Option[HatClaim] = {
+    for {
+      hatName <- refineV[NonEmpty](hatNameAsStr).toOption
+      hatClaim = HatClaim(HatName(hatName), contractId)
+    } yield hatClaim
+  }
 
-    eventuallyResponse.map { response =>
-      response.status match {
-        case OK => {
-          val maybeArrayByte = Json.parse(response.body).asOpt[Array[Byte]]
-          maybeArrayByte match {
-            case None =>
-              Left(PublicKeyRequestFailure.InvalidPublicKeyFailure(
-                "The response was not able to be decoded into an Array[Byte]."))
-            case Some(arrayByte) => Right(PublicKeyReceived(arrayByte))
-          }
-        }
-        case _ => {
-          Left(PublicKeyRequestFailure.ServiceRespondedWithFailure(
-            s"The Adjudicator Service responded with an error: ${response.statusText}"))
-        }
+  private def createHatClaim(maybeHatName: Option[HatName], contractId: ContractId): Option[HatClaim] = {
+    for {
```

Comment:
wondering if a for comp is best here
would  ``` maybeHatName.map(HatClaim(_, contractId)) ``` read better ?
is this repeated enough that it really need a function ?